### PR TITLE
BaSP-TG-10: Add and Edit Tasks

### PR DESCRIPTION
**Pull request by mara666**
- Create add/edit form including: Description
- On edit functionality should show previuos info
- Success message for confirmation edit and add actions (temporarily with alert)
- Redirect to edit/add form
- Warning/error messages for edit and add (temporarily with alert)
- Create add functionality
- Create edit functionality

### CODE REVIEW

File: src/Components/Tasks/form.jsx [hunk 1]
```diff
@@ -1,14 +1,108 @@
...
+import styles from './tasks.module.css';
+
 const Form = () => {
+  const [inputValue, setInputValue] = useState({
+    description: ''
+  });
+
+  useEffect(async () => {
+    const params = new URLSearchParams(window.location.search);
+    const taskId = params.get('id');
```
Comment (DamianPalavecino) [inline src/Components/Tasks/form.jsx:11]: Here you can set the constants 'params' and 'taskId' as global variables so you don't have to repeat them at line 36

File: src/Components/Tasks/form.jsx [hunk 2]
```diff
@@ -1,14 +1,108 @@
...
+      try {
+        const response = await fetch(`${process.env.REACT_APP_API_URL}/tasks/${taskId}`);
+        const data = await response.json();
+        setInputValue({
+          description: data.data.description
+        });
+      } catch (err) {
+        setTimeout(() => {
+          alert('There is no task with id provided');
+        }, 10);
```
Comment (DamianPalavecino) [inline src/Components/Tasks/form.jsx:22]: Remove this useless SetTimeout function

File: src/Components/Tasks/form.jsx [hunk 3]
```diff
@@ -1,14 +1,108 @@
...
+        },
+        body: JSON.stringify(inputValue)
+      };
+      const url = `${process.env.REACT_APP_API_URL}/tasks/${taskId}`;
+      fetch(url, putOptions).then(async (response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          const { message } = await response.json();
+          alert(message);
+        } else {
+          alert('Task was successfully edited');
```
Comment (PilarFernandezC) [inline src/Components/Tasks/form.jsx:52]: Here you could add the message like you did in line 50

File: src/Components/Tasks/form.jsx [hunk 4]
```diff
@@ -1,14 +1,108 @@
...
+        },
+        body: JSON.stringify(inputValue)
+      };
+      const url = `${process.env.REACT_APP_API_URL}/tasks`;
+      fetch(url, postOptions).then(async (response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          const { message } = await response.json();
+          alert(message);
+        } else {
+          alert('Task was created successfully');
```
Comment (PilarFernandezC) [inline src/Components/Tasks/form.jsx:70]: same here

File: src/Components/Tasks/form.jsx (hunk 1 above)
Comment (mara666) [reply src/Components/Tasks/form.jsx]: fixed! Thanks!

File: src/Components/Tasks/form.jsx (hunk 2 above)
Comment (mara666) [reply src/Components/Tasks/form.jsx]: fixed!

File: src/Components/Tasks/form.jsx (hunk 3 above)
Comment (mara666) [reply src/Components/Tasks/form.jsx]: fixed! Thanks!

File: src/Components/Tasks/form.jsx (hunk 4 above)
Comment (mara666) [reply src/Components/Tasks/form.jsx]: fixed

**Review by martumoser — APPROVED** [review]

**Review by PilarFernandezC — APPROVED** [review]

**Pull request merged**